File: PokePlannerWeb.Data/DataStore/Models/StatEntry.cs
using System.Collections.Generic;
using System.Linq;

namespace PokePlannerWeb.Data.DataStore.Models
{
    /// <summary>
    /// Represents a stat in the data store.
    /// </summary>
    public class StatEntry : NamedApiResourceEntry
    {
        /// <summary>
        /// Gets the ID of the stat.
        /// </summary>
        public int StatId => Key;

        /// <summary>
        /// Gets or sets this stat's display names.
        /// </summary>
        public List<LocalString> DisplayNames { get; set; }

        /// <summary>
        /// Gets or sets whether this stat is a battle-only stat.
        /// </summary>
        public bool IsBattleOnly { get; set; }

        /// <summary>
        /// Returns the name of this stat entry in the given locale.
        /// </summary>
        public string GetDisplayName(string locale = "en")
        {
            return DisplayNames.SingleOrDefault(n => n.Language == locale)?.Value;
        }
    }
}

File: PokePlannerWeb/ClientApp/src/models/StatEntry.ts
import { LocalString } from "./LocalString"

/**
 * Represents a stat in the data store.
 */
export class StatEntry {
    /**
     * The stat's ID.
     */
    statId: number

    /**
     * The name of the stat.
     */
    name: string

    /**
     * The stat's display names.
     */
    displayNames: LocalString[]

    /**
     * Whether the stat is a battle-only stat.
     */
    isBattleOnly: boolean

    /**
     * Constructor.
     */
    constructor(
        statId: number,
        name: string,
        displayNames: LocalString[],
        isBattleOnly: boolean
    ) {
        this.statId = statId
        this.name = name
        this.displayNames = displayNames
        this.isBattleOnly = isBattleOnly
    }

    /**
     * Returns a stat created from the given entry.
     */
    static from(stat: StatEntry) {
        return new StatEntry(
            stat.statId,
            stat.name,
            stat.displayNames,
            stat.isBattleOnly
        )
    }

    /**
     * Returns the stat's display name in the given locale.
     */
    getDisplayName(locale: string): string | undefined {
        let localName = this.displayNames.find(n => n.language === locale)
        if (localName === undefined) {
            console.warn(
                `Stat ${this.statId} is missing display name in locale '${locale}'`
            )
        }

        return localName?.value
    }
}